### PR TITLE
feat(elements): SignIn - forgot password

### PR DESCRIPTION
**Pull request by tmilewski**
### Machines/React:

Adds Sign In Forgot Password functionality, 

A relatively small amount of changes here as it leans heavily on the existing Choose Strategy implementation. Most of the code changes lie in applying the updates to the example sign-in page.

```tsx
<Step name='forgot-password'>
  <SupportedStrategy name='reset_password_email_code'>Reset your password via Email</SupportedStrategy>
  {/* ... */}
</Step>

<Step name='reset-password'>
  <Clerk.Field name='password'>
    <Clerk.Label>New Password</Clerk.Label>
    <Clerk.Input />
    <Clerk.FieldError />
  </Clerk.Field>

  <Action submit>Change Password</Action>
  {/* ... */}
</Step>

<Step name='verifications'>
  <Strategy name='reset_password_email_code'>
    <h2>Verify your email</h2>
    <p>We&apos;ve sent a code to <SafeIdentifier /></p>

    <Clerk.Field name='code'>
      <Clerk.Label>Code</Clerk.Label>
      <Clerk.Input />
      <Clerk.FieldError />
    </Clerk.Field>

    <Action submit>Continue</Action>
  </Strategy>
  {/* ... */}
</Step>
```

### React:
Starts the transition to `SupportedStrategy`.

### Example App:
Slims down examples by moving `Submit` loading functionality into abstracted button.

SDK-1138
SDK-1529

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/elements/src/react/sign-in/action/action.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+ *
+ * @example
+ * import { Action } from '@clerk/elements/sign-in';
+ * <Action submit>Sign In</Action>
+ *
+ * @example
+ * import { Action } from '@clerk/elements/sign-in';
+ * <Action resend>Resend</Action>
+ */
+export const SignInAction = React.forwardRef<React.ElementRef<'button'>, SignInActionProps>((props, forwardedRef) => {
```
Comment (BRKalow) [inline packages/elements/src/react/sign-in/action/action.tsx:43]: Are these changes related to the reset password flow? I think not, just clarifying that it's a refactor 👀

**Review by BRKalow — APPROVED** [review]
lgtm!

File: packages/elements/src/react/sign-in/action/action.tsx (hunk 1 above)
Comment (tmilewski) [reply packages/elements/src/react/sign-in/action/action.tsx:43]: It is a part of a refactor. Just moving the files into a more logical file structure. Largely just grouping Steps and Actions.

File: packages/elements/src/internals/machines/sign-in/machines/router.machine.ts [hunk 2]
```diff
@@ -249,14 +256,26 @@ export const SignInRouterMachine = setup({
...
           },
           {
-            actions: ['logUnknownError', { type: 'navigateInternal', params: { path: '/' } }],
+            guard: 'statusNeedsNewPassword',
+            actions: { type: 'navigateInternal', params: { path: '/reset-password' } },
+            target: 'ResetPassword',
           },
         ],
+        'STRATEGY.UPDATE': {
+          description: 'Send event to verification machine to update the current factor.',
```
Comment (LekoArts) [inline packages/elements/src/internals/machines/sign-in/machines/router.machine.ts:265]: Nit: Desc text is not accurate

File: packages/elements/src/internals/machines/sign-in/machines/router.machine.ts [hunk 3]
```diff
@@ -249,14 +256,26 @@ export const SignInRouterMachine = setup({
...
+          actions: sendTo('firstFactor', ({ event }) => event),
+          target: '.Idle',
+        },
       },
       initial: 'Idle',
       states: {
         Idle: {
           on: {
+            'NAVIGATE.FORGOT_PASSWORD': {
+              description: 'Send event to verification machine to update the current factor.',
```
Comment (LekoArts) [inline packages/elements/src/internals/machines/sign-in/machines/router.machine.ts:275]: Nit: Desc text is not accurate

File: packages/elements/src/react/common/loading.tsx [hunk 4]
```diff
@@ -157,15 +156,15 @@ function SignInLoading({ children, scope, routerRef }: SignInLoadingProps) {
...
+  const snapshot = routerRef.getSnapshot();
+  const isFirstFactor = snapshot.hasTag('route:first-factor');
+
   // Determine loading states based on the step
   const isStartLoading = isLoading && loadingStep === 'start';
   const isVerificationsLoading = isLoading && loadingStep === 'verifications';
-  const isChooseStrategyLoading = isLoading && isChooseStrategyStep;
+  const isChooseStrategyLoading = isLoading && isFirstFactor && snapshot.hasTag('route:choose-strategy');
+  const isForgotPasswordLoading = isFirstFactor && snapshot.hasTag('route:forgot-password');
+  const isResetPasswordLoading = isFirstFactor && snapshot.hasTag('route:reset-password');
```
Comment (LekoArts) [inline packages/elements/src/react/common/loading.tsx:167]: Nit: `isLoading` check is missing (since we do that in the other boolean constants, too)

**Review by LekoArts — APPROVED** [review]
